### PR TITLE
feat(ibm-classic): add ability to pass user_data on instance launch

### PR DESCRIPTION
**Pull request by a-dubs**
The Softlayer python sdk apparently accepts a "userdata" kwarg which it passes to the instance as cloud-init userdata. This commit leverages that to allow for launching instances on IBM Classic with cloud-init userdata.

## Testing of changes

These changes were tested using the new `launch_with_basic_cloud_config()` demo function in `examples/ibm_classic.py`

### CODE REVIEW

File: pycloudlib/ibm_classic/cloud.py [hunk 1]
```diff
@@ -218,12 +218,11 @@ def launch(
...
             combined with the disk_size to create the instance flavor. For
             example, B1_2X4 with disk_size of 25G would result in "B1_2X4X25".
-            user_data: Will raise an exception if provided because IBM
-            Classic does not support user data.
+            user_data: cloud-init user data to pass to the instance
             datacenter_region: region to launch the instance in.
             This will automatically select a datacenter in the region if
             "datacenter" is not provided.
-            datacenter: datacenter to launch the instance in. If not
+            datacenter: datacenter to launch the instance in. If notq
```
Comment (TheRealFalcon) [inline pycloudlib/ibm_classic/cloud.py:225]: Nit: typo
```suggestion
            datacenter: datacenter to launch the instance in. If not
```

**Review by TheRealFalcon — COMMENTED** [review]
+1 but looks like you have a test to fix too.

**Comment by a-dubs** [comment]
@TheRealFalcon im gonna be so real - I thought about tests but then was like nahhh no way I wrote any tests for ibm classic and just moved on. Turns out I did! What a good lil developer past me was :P

**Comment by a-dubs** [comment]
Okay wait am I bugging? I don't see any tests that are failing?

**Review by TheRealFalcon — APPROVED** [review]
Weird...didn't see the context, but looks good now, so +1.

**Pull request merged**